backend/notifications/NotificationService.cs

```diff
@@ -163,16 +175,17 @@ public async Task<Model.StatusResponse> GetStatusAsync(Guid messageId)
         private void CompileTemplate<TModel>(string templateKey, Model.IEmailTemplate template)
         {
             var subjectKey = String.Format(SUBJECT_TEMPLATE_KEY, templateKey);
+            var type = typeof(TModel) == typeof(object) || typeof(TModel).IsAnonymousType() ? null : typeof(TModel);
             if (!_cache.ContainsKey(subjectKey))
             {
-                Engine.Razor.Compile(template.Subject, subjectKey, typeof(TModel));
+                Engine.Razor.Compile(template.Subject, subjectKey, type);
```

Comment:
Razor has a memory leak that occurs if a template is compiled and never used again.  Due to complexities noted by the owner of the library the only work around is to generate a new compiled template when a new version is generated.  This however means that if someone updated a template many times, each would be placed into memory until the service was restarted...  This will regrettably be tech-debt and a complex one at that...